pages/api/socket.ts

```diff
@@ -0,0 +1,36 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { Server as IOServer } from 'socket.io';
+import { Socket as NetSocket } from 'net';
+import { Server as HTTPServer } from 'http';
+
+type Res = NextApiResponse & {
+  socket: NetSocket & {
+    server: HTTPServer & {
+      io?: IOServer;
+    };
+  };
+};
+
+export default function handler(
```

Comment:
Are we handling all the backend game logic in this handler?

Reply:
`handler`s in API routes handle everything server request. Here's a good read: https://nextjs.org/docs/api-routes/introduction

As for this route, you are correct. This is where all the socket interaction will happen.